_posts/2024-09-23-wai-aria.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: What is WAI-ARIA and When to Use It
+author: marja
+excerpt: >
+  An introduction to WAI-ARIA and a brief comparison of it with HTML.
+tags:
+ - WAI-ARIA
+ - ARIA
+ - Accessibility
+ - HTML
+---
+
+WAI-ARIA is not directly mentioned in the [WCAG 2.1](https://www.w3.org/TR/WCAG21/) at all. So, what does it have to do with accessibility?  
+
+## What is WAI-ARIA?
+
+WAI-ARIA stands for "Web Accessibility Initiative - Accessible Rich Internet Applications," and is often referred to simply as ARIA. ARIA is written and updated by the [W3C](https://www.w3.org/).
+
+ARIA is a set of rules and attributes that allow developers to define more refined semantics for different UI elements. Often, when JavaScript is added to user interactions, extra guidance is needed for screen readers and other accessibility tools. ARIA enables the use of non-semantic HTML, though this doesn't mean developers should aim to write non-semantic HTML. ARIA rules and attributes help accessibility tools convey to users what is happening on the page and what interactions are possible.
+
+## HTML vs. ARIA: Which Should You Use?
+
+When comparing HTML to ARIA, ARIA should always be a last resort. **If there is a semantic HTML element available, use that.** There’s no need for HTML like this: `<div role="navigation">...</div>`. Instead, you should use a native HTML element like `<nav>...</nav>`.
+
+Before using ARIA, check [the native HTML elements](https://developer.mozilla.org/en-US/docs/Web/HTML) to ensure none apply to your situation. A lot can be achieved with HTML alone.
+
+## How to Use ARIA
+
+As mentioned, ARIA consists of a set of roles and attributes. Let's go over what they are used for.
+
+### Roles
+
+ARIA roles make it possible to describe an element's function in a way that HTML can't. Some HTML elements have default roles. For example, `<input type="radio">` automatically has the role of "radio." Other elements have no role unless one is specifically assigned.
```

Comment:
> Some HTML elements have default roles. For example, `<input type="radio">` automatically has the role of "radio." Other elements have no role unless one is specifically assigned.

To me, this text was hard to grasp at first. I think it would have been easier for me if it were phrased something like this:

> HTML elements have no role unless one is specifically assigned, except for form elements like `<input type="radio">`.

I'm not sure why it feels easier for me to read. Maybe it's because it is a longer sentence with punctuation and flow? I don't know.